aea/configurations/validation.py

```diff
@@ -314,7 +315,20 @@ def check_excludes(path: Tuple[str, ...]) -> bool:
             continue
 
         if path not in pattern_path_value:
-            errors.append(f"Attribute `{'.'.join(path)}` is not allowed to be updated!")
+            is_a_dict_override = False
+            path_ = deepcopy(path)
+            while len(path_) > 0:
+                path_ = path_[:-1]
+                if path_ not in pattern_path_value:
+                    continue
+                pattern_value = pattern_path_value[path_]
+                is_a_dict_override = isinstance(pattern_value, OrderedDict)
+
+            if not is_a_dict_override:
+                errors.append(
+                    f"Attribute `{'.'.join(path)}` is not allowed to be updated!"
+                )
```

Comment:
To explain this change take this override for example

```yaml
...
---
public_id: dummy_author/test_skill:0.1.0
type: skill
models:
  scaffold:
    args:
      recursive: {}
```

Now if run 

```bash
aea config set skills.test_skill.models.scaffold.args.recursive "{\"hello\":\\"World\"}"
```

The path for the override will be parsed as `skills.test_skill.models.scaffold.args.recursive.hello` and since `hello` not an actual param but rather a part of the dict param this will cause the update to fail. The path for `recursive` param looks like this `("models", "scaffold", "args", "recursive")` and path generated by the config manager will be `("models", "scaffold", "args", "recursive", "hello")` so we keep reducing the path until either we find a matching pattern or path tuple is empty.

Reply:
1. What is happening here? You only detect `pattern_value` else reduce path and error if needed. So it's just a check to ensure the path is a valid dict override.

2. I would like to have this refactored into a separate function with appropriate naming. Keep it nested so it remains scoped and such that you don't have to pass the `errors` as an argument to it.
```
if path not in pattern_path_value:
    ensure_valid_dict_override(path)
```

Reply:
Do you think we could update `dict_to_path_value` instead?

Reply:
No we cannot, since we also need to make the path from component config is also a dict param

https://github.com/valory-xyz/open-aea/pull/284/files/b831e33696a54ec0207b5f08e08213c0e1343d3b#diff-df23aead1fcd7e33689e389539d0c8bc863f514d65b8d28767bc747b56dc6834R325